document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreTest.java

```diff
@@ -1687,6 +1710,71 @@ public void testQueryV1AggregationExpression(String dataStoreName) throws IOExce
         dataStoreName, resultDocs, 3, "mongo/test_aggregation_expression_result.json");
   }
 
+  @ParameterizedTest
+  @MethodSource("databaseContextProvider")
+  public void testQueryV1AggregationFilter(String dataStoreName) throws IOException {
+    Map<Key, Document> documents = createDocumentsFromResource("mongo/collection_data.json");
+    Datastore datastore = datastoreMap.get(dataStoreName);
+    Collection collection = datastore.getCollection(COLLECTION_NAME);
+
+    // add docs
+    boolean result = collection.bulkUpsert(documents);
+    Assertions.assertTrue(result);
```

Comment:
Maybe in a separate PR, could we make a method for lines from 1716 to 1722 as adding documents part seems to be repeated.

Reply:
Also, if possible, as an optimisation, we could move all the query tests into a single class with a `setUp()` (@BeforeAll) method that inserts documents just once.

Reply:
Ya, we will have to clean up tests. Planning to do as follow-up.

Reply:
Sure. Makes sense.
In general, for any potential follow-ups, could we just add a single `TODO` comment so that we can keep track?